### ISSUE TITLE
2019/03/29 18:55:40.090235 cmd_run.go:876: WARNING: cannot create user data directory: failed to verify SELinux context of /home/foo/snap: exec: "matchpathcon": executable file not found in $PATH

### ISSUE THREAD

**Issue by andrebengaly** [PowerShell/PowerShell#9252]
# Environment data

2019/03/29 18:55:40.090235 cmd_run.go:876: WARNING: cannot create user data directory: failed to verify SELinux context of /home/foo/snap: exec: "matchpathcon": executable file not found in $PATH

```none

```

**Comment by SpainTrain** [comment]
`matchpathcon` is a command from SELinux utilities (http://manpages.ubuntu.com/manpages/trusty/man8/matchpathcon.8.html).  You probably need to install it.  On Ubuntu : `sudo apt install selinux-utils`

**Comment by andrebengaly** [comment]
2019/04/04 02:22:49.966135 cmd_run.go:876: WARNING: cannot create user data directory: failed to verify SELinux context of /home/foo/snap: exit status 1
PowerShell 6.2.0
Copyright (c) Microsoft Corporation. All rights reserved.

https://aka.ms/pscore6-docs
Type 'help' to get help.

PS /home/foo>

**Comment by colek42** [comment]
having this issue in vscode on ubuntu 18.04.

**Comment by andrebengaly** [comment]
PS /home/foo> uname -a
Linux foo2 5.0.6-050006-generic #201904030534 SMP Wed Apr 3 05:36:14 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
PS /home/foo> lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.10
Release:	18.10
Codename:	cosmic
PS /home/foo>

**Comment by sunzhec2702** [comment]
I also met this issue.

#lxc list
2019/04/22 09:38:08.109128 cmd_run.go:876: WARNING: cannot create user data directory: failed to verify SELinux context of /home/nvidia/snap: exec: "matchpathcon": executable file not found in $PATH

after I install matchpathcon by "sudo apt install selinux-utils", I met below error.

cmd_run.go:876: WARNING: cannot create user data directory: failed to verify SELinux context of /home/nvidia/snap: exit status 1

**Comment by priv** [comment]
Same here with lxc, ubuntu 19.04

**Comment by zelgerj** [comment]
I'am on Ubuntu 18.04, recently switched from kernel 4.20 to 5.06 and had the same problems. After downgrading to kernel 4.20 the problem was gone.

**Comment by bboozzoo** [comment]
The problem is caused by booting a system with a kernel that has SELinux enabled in it's configuration and uses it as a default security backend (CONFIG_DEFAULT_SECURITY), while at the same time the userspace has none or just rudimentary support for SELinux. In other words, the system advertises that it supports SELinux, but is missing the userspace tools and/or policy bits to actually make it work.

AFAIK Ubuntu enables AppArmor as a default (and only) MAC in their kernels, and the userspace is built with similar assumption. It looks to me that the 'mainline' kernel people grab from some random PPA is not ready to be used with such userspace.

**Comment by blinkeye** [comment]
Thanks @bboozzoo  and @zelgerj  for the information. Any ideas for a workaround (without going back to the LTS kernel)?

**Comment by bboozzoo** [comment]
I've described a workaround here: https://forum.snapcraft.io/t/selinux-warning-when-running-lxc/11100/11
TLDR: install `selinux-policy-default`, set `SELINUX=disabled` in `/etc/selinux/config` and reboot the system.

**Comment by zelgerj** [comment]
@bboozzoo Thanks. It works on my side :+1:

**Comment by blinkeye** [comment]
I'm running Ubuntu 18.04, so I've tested with the latest 5.0.9 kernel (still the same warning). To workaround I've updated `/etc/default/grub` and added `GRUB_CMDLINE_LINUX="selinux=0"` and regenerated the grub config with `sudo update-grub` and rebooted: warning gone.

**Comment by TravisEz13** [comment]
Please fill out the issue template completely so that we can investigate.  

I believe this answer is probably correct:  https://github.com/PowerShell/PowerShell/issues/9252#issuecomment-486722749
